libssh2/src/Network/SSH/Client/LibSSH2.hs

```diff
@@ -152,8 +152,8 @@ checkHost :: Session
 checkHost s host port path = do
   kh <- initKnownHosts s
   _numKnownHosts <- knownHostsReadFile kh path
-  (hostkey, _keylen, _keytype) <- getHostKey s
-  result <- checkKnownHost kh host port hostkey [TYPE_PLAIN, KEYENC_RAW]
+  (hostkey, _keytype) <- getHostKey s
+  result <- checkKnownHost kh host port hostkey [TYPE_PLAIN, KEYENC_RAW, KEY_RSA1, KEY_SSHRSA, KEY_SSHDSS]
```

Comment:
I think we also need `KEYENC_BASE64`, since the keys in the known hosts file are base64 encoded.

Reply:
I needed this in my custom implementation at least.

Reply:
I needed this in my implementation at least.

Reply:
Perhaps we should allow the user of this function to pass the flags, and add it as a parameter to the function.

Reply:
I thought maybe it would be better to do
```haskell
(hostkey, keytype) <- getHostKey s
result <- checkKnownHost kh host port hostkey [keytype]
```
what do you think?

Reply:
That is insufficient; the typemask details not just what key type is it, but also what format the host name is and what format the key is in. See 
https://www.libssh2.org/libssh2_knownhost_check.html

For example, we'd pass `[TYPE_PLAIN, KEYENC_BASE64, KEY_SSHRSA]` to specify that the host name is plain (not hashed), that the key is base64 encoded and that the key is an ssh-rsa key.